octree/include/pcl/octree/impl/octree_base.hpp

```diff
@@ -401,9 +401,9 @@ OctreeBase<LeafContainerT, BranchContainerT>::deleteLeafRecursive(
       child_branch = static_cast<BranchNode*>(child_node);
 
       // recursively explore the indexed child branch
-      b_no_children = deleteLeafRecursive(key_arg, depth_mask_arg / 2, child_branch);
+      b_has_children = deleteLeafRecursive(key_arg, depth_mask_arg / 2, child_branch);
 
-      if (!b_no_children) {
+      if (!b_has_children) {
```

Comment:
Is there no unit test covering this if-branch - seems to me the not operator should not be there with the original naming?
Ie if b_no_children is true, the branch node should be removed.

Reply:
The variable is only renamed, but it serves exactly the same logic as before. If `b_no_children == true`, the branch actually has children, and if `!b_no_children == true`, the branch doesn't have children and can be deleted. This PR is to remove this confusing behaviour, if I understood correctly. Important is, that the return value of `branch_arg->hasChild(child_idx)` below is assigned to `b_no_children`/`b_has_children`.

Reply:
Yeah, so it does the opposite of what the original comments said. Ie. It returns false when there is no children, which does fit the logic, just not the naming nor comments.